src/lib/validate-declarative-schema.js

```diff
@@ -122,7 +122,7 @@ const TaskSchema = joi.array().items(
       .error(taskError('"priority" should be an object with optional fields "level" and "label" or a function which returns the same')),
     actions: joi.array().items(
       joi.object({
-        type: joi.string().valid('report', 'contacts').optional(),
+        type: joi.string().valid('report', 'contact').optional(),
         form: joi.string().min(1).required(),
```

Comment:
If we change this to: 
```suggestion
        form: joi.alternatives().conditional('type', { is: 'contact', then: joi.forbidden(), otherwise: joi.string().min(1).required() }),
```
we will make `form` required for `report` action types and forbidden for `contact` action types.